frontend/src/app/Settings/ManageOrganizationContainer.tsx

```diff
@@ -60,30 +71,35 @@ const ManageOrganizationContainer: any = () => {
     return <p>Error: setting not found</p>;
   }
 
-  const onSave = (name: string) => {
+  const onSave = async ({ name, type }: EditableOrganization) => {
     if (appInsights) {
       trackSaveSettings(null);
     }
-    setOrganization({
-      variables: {
-        name,
-      },
-    }).then(() => {
-      let alert = (
-        <Alert
-          type="success"
-          title="Updated Organization"
-          body="The settings for the organization have been updated"
-        />
-      );
-      showNotification(toast, alert);
+    const mutation = isSuperUser
+      ? () => adminSetOrganization({ variables: { name, type } })
+      : () => setOrganization({ variables: { type } });
+
+    const alertProps: ComponentProps<typeof Alert> = {
+      type: "success",
+      title: "Updated Organization",
+      body: "The settings for the organization have been updated",
+    };
+
+    try {
+      await mutation();
```

Comment:
**praise:** Love this anonymous function + ternary combo

Reply:
ty!